File: 9-palindrome-number/9-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0: return False
        y = str(x)
        for i in range(len(y) // 2):
            if y[i] != y[len(y) - i - 1]:
                return False
        return True

File: 9-palindrome-number/9-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        string m = to_string(x);
        int n = m.size();
        int i = 0, j = n - 1;
        while (i <= j) {
            if (m[i] != m[j]) return false;
            ++i;
            --j;
        }
        return true;
    }
};